contracts/SpokePool.sol

```diff
@@ -319,14 +330,18 @@ abstract contract SpokePool is
         address originToken,
         uint256 amount,
         uint256 destinationChainId,
-        uint64 relayerFeePct,
-        uint32 quoteTimestamp
+        int64 relayerFeePct,
+        uint32 quoteTimestamp,
+        uint256 maxCount
     ) public payable override nonReentrant unpausedDeposits {
         // Check that deposit route is enabled.
         require(enabledDepositRoutes[originToken][destinationChainId], "Disabled route");
 
         // We limit the relay fees to prevent the user spending all their funds on fees.
-        require(relayerFeePct < 0.5e18, "invalid relayer fee");
+        require(SignedMath.abs(relayerFeePct) < 0.5e18, "Invalid relayer fee");
+        require(amount < 1e36, "Amount too large");
```

Comment:
why this change?

Reply:
I meant to add the same check on fillRelay. Will add that now.

The idea is that if someone can send an invalid fill of arbitrary size, they could theoretically force the fillCounter to overflow, blocking all fills thereafter (because adding to it would revert). So we want to cap the size of the amount on the fill side to make it infeasible for an attacker to force an overflow on the fill side. We mirror the check on the deposit side to avoid a scenario where a deposit could theoretically be unfillable.

With this cap, someone would need to create 5.78e40 invalid fills to force an overflow, which is effectively infeasible.

Reply:
Done.

Reply:
ok this seems safe